platform/commonUI/browse/src/ObjectHeaderController.js

```diff
@@ -41,25 +41,44 @@ define(
          * @param event the mouse event
          */
         ObjectHeaderController.prototype.updateName = function (event) {
-            if (event && (event.type === 'blur' || event.which === 13)) {
-                var name = event.currentTarget.innerHTML;
+            if (!event) {
+                return;
+            }
+
+            if (this.enterKeyHandled === true) {
+                this.enterKeyHandled = false;
+                return;
+            }
```

Comment:
This approach feels brittle - it will be very sensitive to the particular order of events that come in. This is to avoid hitting an error when `event.currentTarget` is null, is that correct? If so, it would make sense to just check for that directly.

Reply:
Changed the code to check for it directly.